server/src/main/java/org/elasticsearch/rest/action/RestCancellableNodeClient.java

```diff
@@ -113,6 +113,7 @@ private void cancelTask(TaskId taskId) {
     private class CloseListener implements ActionListener<Void> {
         private final AtomicReference<HttpChannel> channel = new AtomicReference<>();
         private final Set<TaskId> tasks = new HashSet<>();
+        private boolean tasksDrained = false;
```

Comment:
NIT: whatever you prefer really but to me it seems that in these spots it's mostly easier to just make `tasks` non-final and `null` ist out to signal `tasksDrained`, removing the need for the copy and somewhat hardening the design against adding a task after tasks have been drained?
It's also one less field which is always nice :)

Reply:
Yeah fair enough, see 20f492f2e13b. Required adding more null checks than I initially expected...

Reply:
Ah right the stats :) thanks!